gdx/src/com/badlogic/gdx/maps/tiled/BaseTmxMapLoader.java

```diff
@@ -467,62 +469,101 @@ protected Cell createTileLayerCell (boolean flipHorizontally, boolean flipVertic
-						byte[] bytes = Base64Coder.decode(data.getText());
-						if (compression == null)
-							is = new ByteArrayInputStream(bytes);
-						else if (compression.equals("gzip"))
-							is = new BufferedInputStream(new GZIPInputStream(new ByteArrayInputStream(bytes), bytes.length));
-						else if (compression.equals("zlib"))
-							is = new BufferedInputStream(new InflaterInputStream(new ByteArrayInputStream(bytes)));
-						else
-							throw new GdxRuntimeException("Unrecognised compression (" + compression + ") for TMX Layer Data");
-
-						byte[] temp = new byte[4];
-						for (int y = 0; y < height; y++) {
-							for (int x = 0; x < width; x++) {
-								int read = is.read(temp);
-								while (read < temp.length) {
-									int curr = is.read(temp, read, temp.length - read);
-									if (curr == -1) break;
-									read += curr;
-								}
-								if (read != temp.length)
-									throw new GdxRuntimeException("Error Reading TMX Layer Data: Premature end of tile data");
-								ids[y * width + x] = unsignedByteToInt(temp[0]) | unsignedByteToInt(temp[1]) << 8
-									| unsignedByteToInt(temp[2]) << 16 | unsignedByteToInt(temp[3]) << 24;
-							}
-						}
-					} catch (IOException e) {
-						throw new GdxRuntimeException("Error Reading TMX Layer Data - IOException: " + e.getMessage());
-					} finally {
-						StreamUtils.closeQuietly(is);
-					}
-				} else {
-					// any other value of 'encoding' is one we're not aware of, probably a feature of a future version of Tiled
-					// or another editor
-					throw new GdxRuntimeException("Unrecognised encoding (" + encoding + ") for TMX Layer Data");
-				}
+			// any other value of 'encoding' is one we're not aware of, probably a feature of a future version of Tiled
+			// or another editor
+			throw new GdxRuntimeException("Unrecognised encoding (" + encoding + ") for TMX Layer Data");
 		}
 		return ids;
 	}
 
+	protected static int[] readIds(InputStream is, int count) throws IOException {
+		int[] ids = new int[count];
+		byte[] temp = new byte[4];
+		for (int idIndex = 0; idIndex < count; idIndex++) {
+			int read = is.read(temp);
+			while (read < temp.length) {
+				int curr = is.read(temp, read, temp.length - read);
+				if (curr == -1) break;
+				read += curr;
+			}
+			if (read != temp.length)
+				throw new GdxRuntimeException("Error Reading TMX Layer Data: Premature end of tile data");
+			ids[idIndex] = unsignedByteToInt(temp[0]) | unsignedByteToInt(temp[1]) << 8
+					| unsignedByteToInt(temp[2]) << 16 | unsignedByteToInt(temp[3]) << 24;
+		}
+		return ids;
+	}
+
+	protected static void loadChunk(
+			int mapWidth, int mapHeight,
+			int chunkXPos, int chunkYPos,
+			int chunkWidth, int chunkHeight,
+			int[] chunkIds, int[] mapIds) {
+		for (int chunkRow = 0; chunkRow < chunkHeight; chunkRow++)
+			for (int chunkCol = 0; chunkCol < chunkWidth; chunkCol++) {
+				int mapCol = chunkCol + chunkXPos;
+				int mapRow = chunkRow + chunkYPos;
```

Comment:
One problem here is that infinite maps can grow in all directions, so `mapCol` and `mapRow` might be negative.

Either the tile layer storage needs to be changed to allow negative coordinates or the entire layer will need to be shifted into positive coordinate space based on the "most negative chunk", possibly shifting it back into position by modifying the layer position (though this is not ideal in terms of game logic, unless this method is applied globally to the map).